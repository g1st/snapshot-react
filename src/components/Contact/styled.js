import styled from 'styled-components';

import {
  colorHeadingText,
  colorText,
  colorMainDark,
  maxContentWidth,
  maxTextWidth
} from '../../styles/variables';

export const Wrapper = styled.div`
  padding: 60px 0 20px 0;
  background-color: ${colorMainDark};
`;
Wrapper.displayName = 'Wrapper';

export const Container = styled.div`
  margin: 0 auto;
  max-width: ${maxContentWidth};
  text-align: center;
`;
Container.displayName = 'Container';

export const Form = styled.form``;
Form.displayName = 'Form';

export const HeadingText = styled.div`
  text-transform: uppercase;
  color: ${colorHeadingText};
  letter-spacing: 6px;
  font-size: 1.8rem;
  line-height: 36px;
  padding: 0 10px 40px 10px;
  font-weight: 300;
  @media (min-width: 768px) {
    letter-spacing: 14px;
    font-size: 2.4rem;
    padding-bottom: 60px;
  }
`;
HeadingText.displayName = 'HeadingText';

export const Text = styled.div`
  color: ${colorText};
  letter-spacing: 3px;
  max-width: ${maxTextWidth};
  margin: 0 auto;
  line-height: 24px;
  font-size: 1rem;
  font-weight: 300;
  padding: 0 10px;
  @media (min-width: 768px) {
    font-size: 1.2rem;
    line-height: 32px;
  }
`;
Text.displayName = 'Text';
