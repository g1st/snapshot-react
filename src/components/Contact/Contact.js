import React from 'react';

import Form from './Form';
import { Wrapper, Container, HeadingText, Text } from './styled';

const Contact = () => {
  return (
    <Wrapper>
      <Container>
        <HeadingText>Send a message</HeadingText>
        <Text>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi quos
          cupiditate temporibus, quisquam dolorem voluptatem modi beatae.
        </Text>
        <Form />
      </Container>
    </Wrapper>
  );
};

export default Contact;
