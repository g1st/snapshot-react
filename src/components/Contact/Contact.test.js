import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Contact from '.';

describe('Contact Component', () => {
  it('matches snapshot', () => {
    const tree = shallow(<Contact />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it('Wrapper must have #contact id attribute', () => {
    const tree = shallow(<Contact />);
    expect(tree.find('Wrapper').find('#contact')).toHaveLength(1);
  });
});
