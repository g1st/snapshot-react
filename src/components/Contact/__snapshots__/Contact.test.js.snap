// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component matches snapshot 1`] = `
<Wrapper
  id="contact"
>
  <Container>
    <HeadingText>
      Send a message
    </HeadingText>
    <Text>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi quos cupiditate temporibus, quisquam dolorem voluptatem modi beatae.
    </Text>
    <Form />
  </Container>
</Wrapper>
`;
